torch/csrc/jit/codegen/cuda/glfdc/expr.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include <torch/csrc/jit/codegen/cuda/glfdc/bitvector.h>
+#include <torch/csrc/jit/codegen/cuda/glfdc/sexpr.h>
+
+#include <c10/util/Exception.h>
+
+#include <map>
+#include <vector>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+namespace glfdc {
+
+struct ExprDAG;
+
+// Expr - points subexpression and its owner expression DAG (ExprDAG)
+// NB: Expr should not outlive ExprDAG, since stores index to;
+// This is convinence class which wraps subexpression and also stores its DAG
+struct Expr {
+  const ExprDAG* dag_; // owner DAG
+  Operand subexpr_; // subexpression refered to inside
+};
+
+// Keeps DAG of all expressions reduction graphs
+struct ExprDAG {
```

Comment:
nit: structs are used only for trivial case in our codebase, so `class` is recommended here.

Reply:
Will do, usually I just picked one that gives me less public/private clauses.